packages/ui5-middleware-livecompileless/lib/index.js

```diff
@@ -0,0 +1,45 @@
+const log = require("@ui5/logger").getLogger("server:custommiddleware:injectless")
+const less = require("less-openui5");
+const {fsInterface} = require("@ui5/fs");
+
+
+/**
+ *
+ * @param {Object} parameters Parameters
+ * @param {Object} parameters.resources Resource collections
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.all Reader or Collection to read resources of the
+ *                                        root project and its dependencies
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.rootProject Reader or Collection to read resources of
+ *                                        the project the server is started in
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.dependencies Reader or Collection to read resources of
+ *                                        the projects dependencies
+ * @param {Object} parameters.options Options
+ * @param {string} [parameters.options.configuration] Custom server middleware configuration if given in ui5.yaml
+ * 
+ * @returns {function} Middleware function to use
+ */
+module.exports = async ({resources, options}) => {
+    const isDebug = options.configuration && options.configuration.debug;
+
+    return async function injectLess(req, res, next) {
+        let url = req.url;
+        if (url.includes(".css") && !url.includes("resources/")) {
+            let possibleLessFile = await resources.rootProject.byPath(url.replace(".css", ".less"));
```

Comment:
w/o double checking at runtime: does this resolve to `false` or `null` in case there's nothing to replace?
https://sap.github.io/ui5-tooling/api/module-@ui5_fs.AbstractReader.html#byPath only mentions a `Promise`

Reply:
The await is unwrapping the promise and when it is not found it will be null